tests/functional/routes/test_account.py

```diff
@@ -243,6 +244,58 @@ def test_change_email_to_existing_email(
 
         email_changed_email.assert_not_called()
 
+    def test_account_delete_with_correct_credentials(self, client, current_user):
+        """DELETE to /account url succeeds and account no longer exists"""
+        res = client.delete("/account")
+        user = User.query.filter_by(uuid=current_user.uuid).first()
+
+        assert res.status_code is 200
+        assert user is None
+
+    def test_account_delete_account_with_reserved_subdomains(
+        self, client, current_user, session
+    ):
+        """DELETE to /account url succeeds and associated reserved subdomains no longer exists"""
+        sub1 = subdomain.ReservedSubdomainFactory(user=current_user, name="subby-sub")
+        sub2 = subdomain.ReservedSubdomainFactory(user=current_user, name="sub-subby")
+        sub3 = subdomain.ReservedSubdomainFactory(user=current_user, name="subwoofer")
+        session.add(sub1)
+        session.add(sub2)
+        session.add(sub3)
+        session.flush()
+
+        assert Subdomain.query.filter_by(id=sub1.id).first() is not None
+
+        delete_res = client.delete("/account")
+
+        assert delete_res.status_code is 200
+        assert User.query.filter_by(uuid=current_user.uuid).first() is None
+        assert Subdomain.query.filter_by(id=sub1.id).first() is None
```

Comment:
It's probably sufficient to verify that the first one has been deleted here without creating the other two.  You don't really use them again.